libpkl/src/main/java/org/pkl/libpkl/LibPklLogger.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2025 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.libpkl;
+
+import java.util.Objects;
+import org.pkl.core.messaging.MessageTransports.Logger;
+
+public class LibPklLogger implements Logger {
+
+  @Override
+  public void log(String msg) {
+    if (Objects.equals(System.getenv("PKL_DEBUG"), "1")) {
+      System.err.println("[libpkl] " + msg);
+    }
```

Comment:
Does this work? What's the preferred approach to take here?

Reply:
You should be able to test this locally (set the env var, run unit tests and see what happens)